src/IB/IBFEMethod.cpp

```diff
@@ -884,34 +928,50 @@ IBFEMethod::computeLagrangianForce(const double data_time)
     batch_vec_ghost_update(d_X_half_vecs, INSERT_VALUES, SCATTER_FORWARD);
     for (unsigned part = 0; part < d_num_parts; ++part)
     {
-        if (d_is_stress_normalization_part[part])
+        if (d_part_is_active[part])
         {
-            computeStressNormalization(*d_Phi_half_vecs[part], *d_X_half_vecs[part], data_time, part);
+            if (d_is_stress_normalization_part[part])
+            {
+                computeStressNormalization(*d_Phi_half_vecs[part], *d_X_half_vecs[part], data_time, part);
+            }
+            d_F_rhs_vecs[part]->zero();
+            assembleInteriorForceDensityRHS(
+                *d_F_rhs_vecs[part], *d_X_half_vecs[part], d_Phi_half_vecs[part], data_time, part);
+        }
+        else
+        {
+            d_F_rhs_vecs[part]->zero();
         }
-        d_F_rhs_vecs[part]->zero();
-        assembleInteriorForceDensityRHS(
-            *d_F_rhs_vecs[part], *d_X_half_vecs[part], d_Phi_half_vecs[part], data_time, part);
     }
     batch_vec_assembly(d_F_rhs_vecs);
     for (unsigned part = 0; part < d_num_parts; ++part)
     {
-        // TODO: Commenting out this line changes the solution slightly but
-        // probably speeds up this solve slightly too.
-        d_F_systems[part]->solution->zero();
-        d_active_fe_data_managers[part]->computeL2Projection(*d_F_systems[part]->solution,
-                                                             *d_F_rhs_vecs[part],
-                                                             FORCE_SYSTEM_NAME,
-                                                             d_use_consistent_mass_matrix,
-                                                             /*close_U*/ false,
-                                                             /*close_F*/ false);
-        *d_F_half_vecs[part] = *d_F_systems[part]->solution;
-        if (d_direct_forcing_kinematics_data[part])
+        if (d_part_is_active[part])
```

Comment:
Let's not include this guard --- I think that I would like for all computations to proceed "as usual" for inactivated parts, except that at the end of the day, forces are not spread and velocity is not interpolated.